mmv1/third_party/terraform/services/databasemigrationservice/resource_database_migration_service_conversion_workspace_test.go

```diff
@@ -0,0 +1,164 @@
+package databasemigrationservice_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDatabaseMigrationServiceConversionWorkspace_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDatabaseMigrationServiceConversionWorkspaceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatabaseMigrationServiceConversionWorkspace_basic(context),
+			},
+			{
+				ResourceName:            "google_database_migration_service_conversion_workspace.example",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"conversion_workspace_id", "location"},
+			},
+		},
+	})
+}
+
+func TestAccDatabaseMigrationServiceConversionWorkspace_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDatabaseMigrationServiceConversionWorkspaceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatabaseMigrationServiceConversionWorkspace_basic(context),
+			},
+			{
+				ResourceName:            "google_database_migration_service_conversion_workspace.example",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"conversion_workspace_id", "location"},
+			},
+			{
+				Config: testAccDatabaseMigrationServiceConversionWorkspace_update(context),
```

Comment:
This needs to check that the resource is being updated in place (not deleted and recreated)

```suggestion
				Config: testAccDatabaseMigrationServiceConversionWorkspace_update(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_database_migration_service_conversion_workspace.example", plancheck.ResourceActionUpdate),
					},
				},
```